src/block_ffm.rs

```diff
@@ -210,6 +234,7 @@ impl<L: OptimizerTrait + 'static> BlockTrait for BlockFFM<L> {
                     let mut ffm_values_offset = 0;
                     for (i, feature) in fb.ffm_buffer.iter().enumerate() {
                         let feature_value = feature.value;
+                        let feature_value_simd = f32x4::splat(feature_value);
```

Comment:
Feature values as such tend to re-appear a lot, did you consider caching the most frequent occurrences? (maybe separate PR)

Reply:
I tried it, but the lookup costs for this are higher then fitting the feature value 4 times.

Reply:
Great to know, thx!